google/cloud/sql/connector/connector.py

```diff
@@ -44,4 +49,8 @@ def connect(instance_connection_string, **kwargs):
     #
     # Return a DBAPI connection
 
-    raise NotImplementedError()
+    loop = asyncio.new_event_loop()
+    thr = Thread(target=loop.run_forever)
+    thr.start()
```

Comment:
2 problems with this: 
1. A reference to Thread isn't being held, so it could be garbage collected
2. We don't want a new Thread/loop for each connect, we want to reuse the same thread/loop for efficiency

You can solve this by using a module level variable:
```python
# This thread is used to background processing
_thread: Optional[Thread] = None
_loop: Optional[asyncio.Loop] = None
```

And a function to memoize:
```python
def _get_loop() -> Thread:
   if _loop is None:
     # init thread and loop
   return _loop
```